charts/influxdb/templates/configmap-backup.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.backup.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "influxdb.fullname" . }}-backup
+  labels:
+    {{- include "influxdb.labels" . | nindent 4 }}
+data:
+  backup.sh: |-
+    #!/bin/bash
+
+    set -e
+
+    DATE="$(date +%Y%m%d_%H%M%S)"
+
+    for DATABASE in $(influx -username {{ .Values.setDefaultUser.user.username }} -password "${INFLUXDB_ADMIN_USER_PASSWORD}" -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc -port {{ .Values.config.http.bind_address | default 8086 }} -execute 'SHOW DATABASES' | sed -e '1,3d'); do
+        echo "backuping ${DATABASE} db to {{ .Values.backup.directory }}/${DATABASE}"
+        mkdir -p {{ .Values.backup.directory }}/${DATABASE}
+        influxd backup -host {{ include "influxdb.fullname" . }}.{{ .Release.Namespace }}.svc:{{ .Values.config.rpc.bind_address | default 8088 }} -portable -db ${DATABASE} {{ .Values.backup.directory }}/"${DATABASE}/${DATE}"
+    done
+
+    echo "deleting old backups"
+    find {{ .Values.backup.directory }} -not -name ".snapshot" -not -name "lost+found" -mindepth 2 -maxdepth 2 -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
+  upload-google.sh: |-
+    #!/bin/sh
+
+    set -ex
+
+    gcloud auth activate-service-account --key-file /var/secrets/google/{{ .Values.backup.uploadProviders.google.secretKey }}
```

Comment:
Please refer to inline script, key file should be optional.

Reply:
Why?
Imho having it in the configmap is much cleaner as blowing up the cronjob with bash scripting.
In the end it becomes unreadable in the cronjob file.

How can the keyfile be optional?
Without it gcloud can never access the storage bucket.
The command also fails without it. See:

```
gcloud auth activate-service-account
ERROR: (gcloud.auth.activate-service-account) argument --key-file: Must be specified.
Usage: gcloud auth activate-service-account [ACCOUNT] --key-file=KEY_FILE [optional flags]
  optional flags may be  --help | --password-file | --prompt-for-password

For detailed information on this command and its flags, run:
  gcloud auth activate-service-account --help
```

Reply:
Sorry I was inaccurate, if you refer to the inline script, the entire gcloud auth activate command is optional, based on wether or not the key is defined. Please see https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity for more info.

While I agree not blowing up the cronjob with 3 different bash scripts is better, it is harder to detect changes when moving to a new file + making changes. Preferably we are only doing one of the two (either moving the scripts, or changing the scripts) for my tired eyes :D

Reply:
tl;dr if we keep the bash scripts in the cornjob it will be easier for review, once the scripts are agreed upon we can move to a configmap?

Reply:
When possible please convert the changes inline so we can properly check the nuanced differences (e.g. another one im saying is DST_URL is replaced with bucket name, but we might want a specific folder in the bucket targetted, etc.) rather than in separate configmap.